File: src/backend/FlatShareBackend/Domain/Models/Booking.cs
namespace FlatShareBackend.Domain.Models;

public class Booking
{
    public required Guid Id { get; set; }
    public required Guid TenantId { get; set; }
    public required Guid ListingId { get; set; }
    public required BookingStatus Status { get; set; }
    public required DateOnly Since { get; set; }
    public required DateOnly Until { get; set; }
    public required decimal TotalCost { get; set; }
    public required string Currency { get; set; }
    public Payment? Payment { get; set; }
}

File: src/frontend/src/types/booking.ts
export const BookingStatus = {
  PendingApproval: 'PENDING_APPROVAL',
  PendingPayment: 'PENDING_PAYMENT',
  Confirmed: 'CONFIRMED',
  Rejected: 'REJECTED',
  Expired: 'EXPIRED',
  PaymentFailed: 'PAYMENT_FAILED',
  Cancelled: 'CANCELLED',
} as const

export type BookingStatus = (typeof BookingStatus)[keyof typeof BookingStatus]


export const PaymentMethod = {
  Card: 0,
} as const

export type PaymentMethod = (typeof PaymentMethod)[keyof typeof PaymentMethod]
export type CreateBookingPayload = {
  listingId: string
  startDate: string
  endDate: string
}

export type BookingCreatedResponse = {
  bookingId: string
  status: BookingStatus
  createdAt?: string
  totalPrice: number
  currency: string
  resourceLink?: string
}

export type CancelBookingPayload = {
  reason: string
}

export type RejectBookingPayload = {
  reason: string
}

export type PayBookingPayload = {
  paymentMethod: PaymentMethod
  returnUrl: string
  cancelUrl: string
}

export type Booking = {
  id: string
  tenantId: string
  listingId: string
  status: BookingStatus
  since: string
  until: string
  totalCost: number
  currency: string
  paymentId?: string | null
}